tools/integration_tests/multiwriters_and_multimounts/multiwriters_from_two_mounts_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multiwriters_and_multimounts_test
+
+import (
+	"os"
+	"path"
+	"strings"
+	"syscall"
+	"testing"
+
+	operations "github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+	. "github.com/jacobsa/ogletest"
+)
+
+func getMountPath1() string {
+	return path.Join(setup.MntDir(), MountName1)
+}
+
+func getMountPath2() string {
+	return path.Join(setup.MntDir(), MountName2)
+}
+
+func getFilePathInMount1() string {
+	return path.Join(getMountPath1(), CommonFileName)
+}
+
+func getFilePathInMount2() string {
+	return path.Join(getMountPath2(), CommonFileName)
+}
+
+func getGCSObjPath() string {
+	return path.Join(setup.TestBucket(), CommonFileName)
+}
+
+func openFile(t *testing.T, filePath string) (file *os.File) {
+	file, err := os.OpenFile(filePath, os.O_RDWR|syscall.O_DIRECT, operations.FilePermission_0600)
+	if err != nil {
+		t.Errorf("error in opening file %s: %v", file.Name(), err)
+	}
+	return
+}
+
+func writeToFile(t *testing.T, file *os.File, fileContent string) {
+	_, err := file.WriteString(fileContent)
+	if err != nil {
+		t.Errorf("error in writing to the file %s: %v", file.Name(), err)
+	}
+}
+
+func syncFile(t *testing.T, file *os.File) {
+	err := file.Sync()
+	if err != nil {
+		t.Errorf("error in syncing file %s: %v", file.Name(), err)
+	}
+}
+
+func cleanAndCreateCommonFileInTestBucket(t *testing.T) {
+	// Cleaning one mount is sufficient as the test bucket is common.
+	err := operations.EmptyDir(getMountPath1())
+	if err != nil {
+		t.Errorf("Error while cleaning mount %v", err)
+	}
+
+	// Create file in mount 1.
+	_, err = os.Create(getFilePathInMount1())
+	if err != nil {
+		t.Errorf("Error creating common file %v", err)
+	}
+	// List files in mount 2 so that the common file starts reflecting in mount 2.
+	_, _ = os.ReadDir(getMountPath2())
+}
+
+func compareGCSObjContent(t *testing.T, objContent string) {
+	tempFile, err := os.CreateTemp(setup.MntDir(), "")
```

Comment:
Why do we need to download fileContents to disk. why can't we read it into memory?